File: app/src/main/java/com/anadi/weatherinfo/mainactivity/MainActivityContract.java
package com.anadi.weatherinfo.mainactivity;

import android.content.Context;
import com.anadi.weatherinfo.repository.LocationInfo;

import java.util.ArrayList;
import java.util.Observer;

public interface MainActivityContract {

    interface View {
        void onUpdate();
    }

    interface Presenter extends Observer {
        void loadLocations(Context context);
        void saveData(Context context);
        void loadData(Context context);

        void subscribe();
        void unsubscribe();

        ArrayList<LocationInfo> getLocations();

        boolean deleteLocation(LocationInfo locationInfo);
    }

    interface Model {
        void loadLocations(Context context);
        void saveData(Context context);
        void loadData(Context context);

        ArrayList<LocationInfo> getLocations();

        boolean deleteLocation(LocationInfo locationInfo);
    }
}

File: app/src/main/java/com/anadi/weatherinfo/ui/mainactivity/MainActivityContract.kt
package com.anadi.weatherinfo.ui.mainactivity

import android.content.Context
import com.anadi.weatherinfo.repository.LocationInfo
import java.util.*

interface MainActivityContract {
    interface View {
        fun onUpdate()
    }

    interface Presenter : Observer {
        var view: View

        fun loadLocations(context: Context)
        fun saveData(context: Context)
        fun loadData(context: Context)
        fun subscribe()
        fun unsubscribe()
        val locations: ArrayList<LocationInfo>
        fun deleteLocation(locationInfo: LocationInfo): Boolean
    }

    interface Model {
        fun loadLocations(context: Context)
        fun saveData(context: Context)
        fun loadData(context: Context)
        val locations: ArrayList<LocationInfo>
        fun deleteLocation(locationInfo: LocationInfo): Boolean
    }
}